ambry-replication/src/main/java/com.github.ambry.replication/ReplicationManager.java

```diff
@@ -377,6 +377,25 @@ public void start() throws ReplicationException {
   }
 
   /**
+   * Enables/disables {@code types} of replication on the given {@code id}s.
+   * @param ids the {@link PartitionId}s to enable/disable it on.
+   * @param origins the list of datacenters from which replication should be enabled/disabled.
+   * @param enable whether to enable ({@code true}) or disable.
+   */
+  public boolean controlReplicationForPartitions(List<? extends PartitionId> ids, List<String> origins,
```

Comment:
Marked down is an internal state of frontends and servers. This is triggered via a tool which doesn't have that notion. So if it is able to connect to the machine and talk to it, then the return will be based on the success/failure of the operation. A storage node being marked down by frontends/other storage nodes does not play a role